src/main/java/taxi/web/servlets/LoginController.java

```diff
@@ -0,0 +1,40 @@
+package taxi.web.servlets;
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import taxi.exception.AuthenticationException;
+import taxi.injections.Injector;
+import taxi.model.Driver;
+import taxi.security.AuthenticationService;
+
+public class LoginController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("taxi");
+    private final AuthenticationService authenticationService =
+            (AuthenticationService) injector.getInstance(AuthenticationService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/login.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+
+        try {
+            Driver driver = authenticationService.login(login, password);
+            req.getSession().setAttribute("driver_id", driver.getId());
+        } catch (AuthenticationException e) {
+            req.setAttribute("message", e.getMessage());
+            req.getRequestDispatcher("/WEB-INF/views/login.jsp").forward(req, resp);
+            return;
```

Comment:
```suggestion
```

Reply:
Why?

Reply:
And why do you need it? Does this return change smth?

Reply:
Nvm, found an explanation article, it's ok 👍